tests/base_conftest.py

```diff
@@ -55,16 +56,20 @@ def __init__(self, classic_contract, method_class=VyperMethod):
         self._classic_contract = classic_contract
         self.address = self._classic_contract.address
         protected_fn_names = [fn for fn in dir(self) if not fn.endswith("__")]
-        for fn_name in self._classic_contract.functions:
-            # Override namespace collisions
-            if fn_name in protected_fn_names:
-                raise AttributeError(f"{fn_name} is protected!")
-            else:
-                _classic_method = getattr(self._classic_contract.functions, fn_name)
-                _concise_method = method_class(
-                    _classic_method, self._classic_contract._return_data_normalizers
-                )
-            setattr(self, fn_name, _concise_method)
+
+        try:
+            for fn_name in self._classic_contract.functions:
+                # Override namespace collisions
+                if fn_name in protected_fn_names:
+                    raise AttributeError(f"{fn_name} is protected!")
+                else:
+                    _classic_method = getattr(self._classic_contract.functions, fn_name)
+                    _concise_method = method_class(
+                        _classic_method, self._classic_contract._return_data_normalizers
+                    )
+                setattr(self, fn_name, _concise_method)
+        except web3.exceptions.NoABIFunctionsFound:
```

Comment:
it's clearer to scope this to where the exception can be raised -- i think just line 61

Reply:
Do you mean something like this?
```
try:
    # Call for side effects
    hasattr(self._classic_contract.functions, "_functions")
except web3.exceptions.NoABIFunctionsFound:
    return

for fn_name in self._classic_contract.functions:
    # Override namespace collisions
    if fn_name in protected_fn_names:
        raise AttributeError(f"{fn_name} is protected!")
    else:
        _classic_method = getattr(self._classic_contract.functions, fn_name)
        _concise_method = method_class(
            _classic_method, self._classic_contract._return_data_normalizers
        )
    setattr(self, fn_name, _concise_method)
```

Reply:
Yes but for cleanliness you can skip the hasattr check and just assign to a variable.
```
try:
  fns = self....functions
except:
  ...
for fn_name in fns:
  ...
```